src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -77,15 +77,13 @@ class BlockchainImpl(
     protected val transactionMappingStorage: TransactionMappingStorage,
     protected val appStateStorage: AppStateStorage,
     protected val stateStorage: StateStorage,
-    blockchainReader: BlockchainReader,
-    blockchainMetadata: BlockchainMetadata
+    blockchainReader: BlockchainReader
 ) extends Blockchain
     with Logger {
 
   override def getChainWeightByHash(blockhash: ByteString): Option[ChainWeight] = chainWeightStorage.get(blockhash)
 
-  override def getLatestCheckpointBlockNumber(): BigInt =
-    blockchainMetadata.bestKnownBlockAndLatestCheckpoint.get().latestCheckpointNumber
+  override def getLatestCheckpointBlockNumber(): BigInt = appStateStorage.getLatestCheckpointBlockNumber()
```

Comment:
if `getBestBlockNumber()` moved to the `blockchainReader`, then I think `getLatestCheckpointBlockNumber()` should also go there

Reply:
You are right. I created this ticket https://jira.iohk.io/browse/ETCM-1092